dask_cuda/tests/test_dask_cuda_worker.py

```diff
@@ -236,3 +241,28 @@ def get_visible_devices():
             del os.environ["CUDA_VISIBLE_DEVICES"]
         if init_nvmlstatus:
             os.environ["DASK_DISTRIBUTED__DIAGNOSTICS__NVML"] = init_nvmlstatus
+
+
+def test_cuda_visible_devices_uuid(loop):  # noqa: F811
+    gpu_uuid = get_gpu_uuid_from_index(0)
+    os.environ["CUDA_VISIBLE_DEVICES"] = gpu_uuid
+    try:
+        with popen(["dask-scheduler", "--port", "9359", "--no-dashboard"]):
+            with popen(
+                [
+                    "dask-cuda-worker",
+                    "127.0.0.1:9359",
+                    "--host",
+                    "127.0.0.1",
+                    "--no-dashboard",
+                    "--worker-class",
+                    "dask_cuda.utils.MockWorker",
+                ]
+            ):
+                with Client("127.0.0.1:9359", loop=loop) as client:
+                    assert wait_workers(client, n_gpus=1)
+
+                    result = client.run(lambda: os.environ["CUDA_VISIBLE_DEVICES"])
+                    assert list(result.values())[0] == gpu_uuid
+    finally:
+        del os.environ["CUDA_VISIBLE_DEVICES"]
```

Comment:
Use `unittest.mock.patch` to avoid deleting `os.environ["CUDA_VISIBLE_DEVICES"]` if it already exist on the system.
Python [docs](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch.dict) has this example:
```python
import os
with patch.dict('os.environ', {'newkey': 'newvalue'}):
    print(os.environ['newkey'])
assert 'newkey' not in os.environ
```

Reply:
That's really cool, I wasn't aware of that. Done in https://github.com/rapidsai/dask-cuda/pull/780/commits/e1070110fa9924dcb9d923f95e6acb8f28246f55 . We have several other tests that can benefit from this, I'll update those in a follow-up PR.